ui/src/components/category/CategoryForm.tsx

```diff
@@ -0,0 +1,107 @@
+"use client";
+
+import { useForm } from "react-hook-form";
+import { Button, TextField, MenuItem, Select, FormControl, FormLabel, FormHelperText } from "@mui/material";
+import { useState } from "react";
+
+interface CategoryFormData {
+    name: string;
+    currency: string;
+}
+
+interface CategoryFormProps {
+    userEmail: string;
+    onCategoryAddedAction: () => void;
+}
+
+const currencies = [
```

Comment:
create or move currencies list to constatns.ts if exists, if not create in shared folder